### PR TITLE
Fix an edge case when the programming exercise is inactive during exams

### PR DESCRIPTION
**Pull request by krusche**
### Checklist
- [ ] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
When the build plan was cleaned up, the programming exercise should have been started within the exam. However this led to an issue displaying the "resume exercise" button. Clicking on it killed the client and led to 20k issues on Sentry and a horrible(!!!) experience for students 👎 

### Description
Fix Fix Fix

### Steps for Testing
1. Create an exam with at least 2 exercises with exact order: 1. programming (only IDE offline activated), 2. text
2. Generate student exams, prepare exercise start
3. Wait 1 week ;-) or cleanup all participations (i.e. delete the build plans within Artemis so that the PESP is displayed as inactive)
4. Switch to the student account, start the exam in the exam mode. Navigate between exercises and verify that no "Resume exercise" button is displayed and that the clone repository button is displayed properly.

The previous behavior (you can verify it on `develop`) showed a resume exercise button, if you click on it, you cannot navigate between exercises any more, because of https://sentry.io/organizations/ls1intum/issues/1988505855/?environment=prod&project=1440029&query=is%3Aunresolved&sort=freq&statsPeriod=14d

You can also see this issue in the following video, proudly provided by one of our nice students (thank you so much!!!)

https://user-images.githubusercontent.com/744067/108558707-9741e280-72fa-11eb-9aba-775ceec8a1cf.mov

### CODE REVIEW

**Review by EneaGore — COMMENTED** [review]
Replicated issue on develop, than tried it out on ts1, unfortunately i get the same issue.


https://user-images.githubusercontent.com/73840596/108581047-a17dd400-732e-11eb-9eec-5dd9878366ad.mp4

**Comment by krusche** [comment]
@EneaGore thanks for testing, then we have to investigate this in more detail again

**Review by airelawaleria — APPROVED** [review]
tested in the testing session with @JohannesWeiss @simonzilker @marvinchimeka 
Maybe we did something wrong, but both the clone button and the switching between the exercises worked for us on ts1

**Review by JohannesWeiss — APPROVED** [review]
Tested in testing session on ts1. For us the issue did not occur.

**Review by simonzilker — APPROVED** [review]
Looks good :)

**Review by marvinchimeka — APPROVED** [review]
No issues found

**Review by ivanchimeno — APPROVED** [review]
Looks good to me!

**Review by philgabr — APPROVED** [review]
For me it workes as described, I could not replicate the error

**Review by SimonEntholzer — APPROVED** [review]
The issue is fixed by this PR :D. I could reproduce the issue locally on develop, but not on the branch with the fix.

A difference I noticed was that on develop multiple submissions were created when a exercise with cleaned up build plan was started.
image
After I clicked resume exercise the stated bug occurs.

In the version of this branch only 1 submission is created, and everything works.
image
 The thing I noticed too, is that a new submission+result is created only for the user that had the build plan cleaned but I'm pretty sure that's not connected to the issue.
image

**Review by EneaGore — APPROVED** [review]
Worked perfectly after the last commit :)

**Review by JannisBittlmayer — APPROVED** [review]
Works as intended on TS1

**Review by FlorianStang — APPROVED** [review]
Tested on TS1 with @JannisBittlmayer, everything worked as intended.

File: src/main/webapp/app/exam/participate/exam-participation.component.ts [hunk 1]
```diff
@@ -546,40 +560,46 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
...
+            if (exercise.studentParticipations) {
+                exercise.studentParticipations!.forEach((participation) => {
+                    if (participation.submissions) {
+                        participation.submissions
+                            .filter((submission) => !submission.isSynced)
+                            .forEach((unsynchedSubmission) => {
+                                submissionsToSync.push({ exercise, submission: unsynchedSubmission });
+                            });
+                    }
+                });
```
Comment (derLalla) [inline src/main/webapp/app/exam/participate/exam-participation.component.ts:576]: this makes it less code.

```suggestion
                exercise.studentParticipations!.forEach((participation) => {
                participation
                    .submissions?.filter((submission) => !submission.isSynced)
                    .forEach((unsynchedSubmission) => {
                        submissionsToSync.push({ exercise, submission: unsynchedSubmission });
                    });
            });
```

**Review by derLalla — CHANGES_REQUESTED** [review]
the code changes look good so far. I have just added a small improvement.

**Pull request merged**